infra/vsts_agent_ubuntu_20_04_startup.sh

```diff
@@ -117,7 +117,14 @@ reset_cache() {
             echo "Killing $pid..."
             kill -s KILL $pid
         done
-        umount $mount_point
+        for pid in $(lsof $mount_point | sed 1d | awk '{print $2}' | sort -u); do
+            echo "Killing $pid..."
+            kill -s KILL $pid
+        done
+        if mount -l | grep $mount_point; then
```

Comment:
When do we expect this to fail?

Reply:
> When do we expect this to fail?

I'm not sure. I don't understand why we get into that state on macOS nodes sometimes. Currently my best guess is that a build gets cancelled in the middle of this script, i.e. it has unmounted the partition already but hasn't mounted the new one yet.

Under that assumption, this could happen on Linux too.